cms/djangoapps/contentstore/views/tests/test_transcripts.py

```diff
@@ -766,3 +766,82 @@ def test_fail_for_non_video_module(self):
         resp = self.client.get(link, {'data': json.dumps(data)})
         self.assertEqual(resp.status_code, 400)
         self.assertEqual(json.loads(resp.content).get('status'), 'Transcripts are supported only for "video" modules.')
+
+
+class TestSaveTranscripts(BaseTranscripts):
+    """
+    Tests for '/transcripts/save' url.
+    """
+    def save_subs_to_store(self, subs, subs_id):
+        """
+        Save transcripts into `StaticContent`.
+        """
+        filedata = json.dumps(subs, indent=2)
+        mime_type = 'application/json'
+        filename = 'subs_{0}.srt.sjson'.format(subs_id)
+
+        content_location = StaticContent.compute_location(self.course.id, filename)
+        content = StaticContent(content_location, filename, mime_type, filedata)
+        contentstore().save(content)
+        del_cached_content(content_location)
+        return content_location
+
+    def assertCurrentSubs(self, expected_subs):
+        """
+        Asserts the current subtitles set on the video module.
+
+        Arguments:
+            expected_subs: expected subtitles which should make our assertion happy.
+        """
+        item = modulestore().get_item(self.video_usage_key)
+        self.assertEqual(item.sub, expected_subs)
+
+    def test_prioritize_youtube_sub_on_save(self):
+        """
+        Test that the '/transcripts/save' endpoint prioritises youtube subtitles over html5 ones
+        while deciding the current subs for video module.
+        """
+        # Update video module to contain 1 youtube and 2 html5 sources.
+        yt_id = str(uuid4())
+        self.item.data = textwrap.dedent(
+            """
+            <video youtube="1:{yt_id}" sub="">
+                <source src="http://www.testvid.org/html5/videos/testvid.mp4"/>
+                <source src="http://www.testvid2.org/html5/videos/testvid2.webm"/>
+            </video>
+            """.format(yt_id=yt_id)
+        )
+        modulestore().update_item(self.item, self.user.id)
+
+        # Assert video module's current subtitles.
+        self.assertCurrentSubs(expected_subs='')
+
+        # Save new subs in the content store.
+        subs = {
+            'start': [100, 200, 240],
+            'end': [200, 240, 380],
+            'text': [
+                'subs #1',
+                'subs #2',
+                'subs #3'
+            ]
+        }
+        self.save_subs_to_store(subs, yt_id)
+
+        # Now, make request to /transcripts/save endpoint with new subs.
+        data = {
+            'locator': unicode(self.video_usage_key),
+            'metadata': {
+                'sub': yt_id
+            }
+        }
+        link = reverse('save_transcripts')
```

Comment:
if we are not using the `link` anywhere else I think we should just pass to this to `client.get` method.